albert-masats/precurso/tema1/calculator-pro.js

```diff
@@ -0,0 +1,143 @@
+/*He hecho un poco de mezcla entre el enunciado que había antes y el que hay ahora. 
+* Se puede pasar cualquier cantidad de números por el prompt y realiza los cálculos mediante funciones raíz cuadrada, suma, resta, mult, div
+* y al final pregunta si queremos repetir la operación.*/
+
+function calculatorPro() { 
+    let numberList = [];
+    let newNumber = null;
+    let acc = 0;
+    const outputResults = [];
+
+    do{
+
+        newNumber = prompt('Please enter number or press cancel to stop');
+        numberList.push(Number(newNumber));    //Convierto los strings a valores numericos al mismo tiempo que los pusheo a numberList.
+
+    }while(newNumber !== null);
+    
+    if (!isFinite(newNumber)) {  
+        alert('Introduce numbers only');
+
+    }else if (newNumber === undefined) {
+        alert('Introduce numbers only');
+
+    }else if (newNumber === null) {
+        alert("Stopped by user");
+    }
+
+    numberList.pop();    //Para quitar el null final del Array que generamos al darle a Cancelar al prompt.
+
+    numberList = numberList.filter( value => !isNaN(value));    //Filtra los NaN.
+    numberList = numberList.filter( value => value !== undefined);    //Filtra los undefined.
+
+    if (!isFinite(numberList[1])) {    //Hacemos la raíz cuadrada cuando solo hay un numero en todo el Array.
+        function squareRoot() {
+           
+            const squareRoot = Math.sqrt(numberList[0]);
+            outputResults.push(squareRoot.toFixed(3));
+            console.log(`Ouptut results = [square root = ${outputResults}]`);
+        }
+        squareRoot();
+
+    }else{
+        function sum(numberList) {    //Si hay más de un dato en el array pasa por las funciones de suma, resta, mult, div.
```

Comment:
Better refactor this function. Take out the function sum and call it before the else instead.